utils/testtools/crypto.go

```diff
@@ -135,72 +46,82 @@ func CreateCSR(key crypto.PrivateKey) (csr []byte, err error) {
 	return csr, nil
 }
 
-// GetCACertificate returns CA certificate.
-func GetCACertificate() (cert []byte) {
-	return []byte(caCertificate)
-}
-
-// GetCAKey returns CA key.
-func GetCAKey() (cert []byte) {
-	return []byte(caKey)
-}
-
-// CreateCertificate creates certificate.
-func CreateCertificate(workDir string, csr []byte) (cert []byte, err error) {
-	tmpDir, err := ioutil.TempDir(workDir, "cert*")
+// GenerateCertifcateAndKeys generate certificate and keys.
+func GenerateCertifcateAndKeys(
+	template, parent *x509.Certificate, privateKey *rsa.PrivateKey,
+) (*x509.Certificate, []byte, *rsa.PrivateKey, error) {
```

Comment:
1. Use named return if function returns more than 2 values: I've updated kb with rules how to use named return values.
2. No need to return values that can be get from each other with some API: *x509.Certificate can be obtained from PEM format by cryptutils.PEMToX509Cert
3. Remove generating private key from this function: it should be separate functions:
- `GenerateCertificate` which takes key as param;
- `GenerateKeyAndCertificate` wich generates key and use `GenerateCertificate`.